File: src/test/java/com/binance/api/domain/market/CandlestickDeserializerTest.java
package com.binance.api.domain.market;

import com.binance.api.client.domain.market.Candlestick;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.fail;

/**
 * Tests the adequate deserialization of candlestick JSON information.
 */
public class CandlestickDeserializerTest {

  @Test
  public void testCandlestickDeserializerTest() {
    final String candlestickJson = "[\n" +
        "    1499040000000,\n" +
        "        \"0.01634790\",\n" +
        "        \"0.80000000\",\n" +
        "        \"0.01575800\",\n" +
        "        \"0.01577100\",\n" +
        "        \"148976.11427815\",\n" +
        "        1499644799999,\n" +
        "        \"2434.19055334\",\n" +
        "        308,\n" +
        "        \"1756.87402397\",\n" +
        "        \"28.46694368\",\n" +
        "        \"17928899.62484339\"\n" +
        "        ]";
    ObjectMapper mapper = new ObjectMapper();
    try {
      Candlestick candlestick = mapper.readValue(candlestickJson, Candlestick.class);
      assertEquals((long)candlestick.openTime, 1499040000000L);
      assertEquals(candlestick.open, "0.01634790");
      assertEquals(candlestick.high, "0.80000000");
      assertEquals(candlestick.low, "0.01575800");
      assertEquals(candlestick.close, "0.01577100");
      assertEquals(candlestick.volume, "148976.11427815");
      assertEquals((long)candlestick.closeTime, 1499644799999L);
      assertEquals(candlestick.quoteAssetVolume, "2434.19055334");
      assertEquals((long)candlestick.numberOfTrades, 308L);
      assertEquals(candlestick.takerBuyBaseAssetVolume, "1756.87402397");
      assertEquals(candlestick.takerBuyQuoteAssetVolume, "28.46694368");
    } catch (IOException e) {
      fail();
    }
  }
}

File: src/test/java/com/binance/api/domain/market/CandlestickDeserializerTest.scala
package com.binance.api.domain.market

import com.binance.api.client.domain.market.Candlestick
import io.circe._
import org.junit.Assert.assertEquals
import org.junit.Test
import com.binance.api.client.json.Decoders._

/**
  * Tests the adequate deserialization of candlestick JSON information.
  */
class CandlestickDeserializerTest {
  @Test def testCandlestickDeserializerTest(): Unit = {
    val json = """[
    1499040000000,
        "0.01634790",
        "0.80000000",
        "0.01575800",
        "0.01577100",
        "148976.11427815",
        1499644799999,
        "2434.19055334",
        308,
        "1756.87402397",
        "28.46694368",
        "17928899.62484339"
        ]"""
    assertEquals(
      Right(
        Candlestick(
          openTime = 1499040000000L,
          open = "0.01634790",
          high = "0.80000000",
          low = "0.01575800",
          close = "0.01577100",
          volume = "148976.11427815",
          closeTime = 1499644799999L,
          quoteAssetVolume = "2434.19055334",
          numberOfTrades = 308,
          takerBuyBaseAssetVolume = "1756.87402397",
          takerBuyQuoteAssetVolume = "28.46694368"
        )
      ),
      parser.decode[Candlestick](json)
    )
  }
}
